### ISSUE TITLE
KML file downloaded contains only a fragment of locations in history

### ISSUE THREAD

**Issue by lefetmeofefet** [yjlou/2019-nCov#13]
Hello, when downloading the kml file i noticed that it's a lot smaller, so i parsed it and it got 340 locations for me, when [here ](https://takeout.google.com/settings/takeout)i get thousands.

**Comment by yjlou** [comment]
Thanks for pointing this out. Actually our tool accepts the Google Takeout
JSON file as well. So feel free to drag&drop your files to compare.

Did you compare the difference between them? As far as I can tell, the data
exported from Timeline and Takeout are similar (of course in different
format). One of them (Timeline I remember) is more *compressed* because
Google would guess your activities and aggregate them. For example, if you
are taking a train, in Takeout you may see more points than Timeout because
Timeline would merge points into a LineString object.

Numbnut <notifications@github.com> 於 2020年3月18日 週三 上午2:43寫道：

**Comment by lefetmeofefet** [comment]
I compared the difference, and you're right, the non-takeout is more about summary of activity, and does not include locations every few seconds. And that's very bad, because when comparing locations you need to know every few seconds where a person was, not hours.
Just for comparison, in one month in the kml file I had 130 locations and in Google takeout I had 12,000. 
I think you should disable the kml option. Also, how did you get the infected people's timelines? Is it by kml link or takeout?

**Comment by yjlou** [comment]
I agree with your point. However, the Takeout UX is even harder for user.
You have to direct user to the Takeout page, have them to click the right
item. Then wait in the gmail for the archive file (undetermined time, could
be hour). Then unzip it and drag/drop to our tool.

Numbnut <notifications@github.com> 於 2020年3月18日 週三 下午6:01寫道：

**Comment by lefetmeofefet** [comment]
I know!! Its terrible. The problem is that right now it's kinda useless with location every few hours.

I also made an app like yours (http://virally.herokuapp.com/) and I spent a lot of time trying to work around this problem. To further the ux problem, Google takeout don't allow iframes :(
I'm thinking the only way to fix this is using an app with a webview that programmatically clicks inside the webview. So sad
